src/api/chassis/model/hDriveModel.cpp

```diff
@@ -144,6 +144,42 @@ void HDriveModel::arcade(const double iforwardSpeed, const double iyaw, const do
   middleMotor->moveVelocity(0);
 }
 
+void HDriveModel::curvature(const double iforwardSpeed, 
+                            const double icurvature, 
+                            const double ithreshold) {
+  // This code is adapted from WPIlib. All credit goes to them. Link:
+  // https://github.com/wpilibsuite/allwpilib/blob/main/wpilibc/src/main/native/cpp/drive/DifferentialDrive.cpp#L49
+  double forwardSpeed = std::clamp(iforwardSpeed, -1.0, 1.0);
+  if (std::abs(forwardSpeed) < ithreshold) {
+    forwardSpeed = 0;
+  }
+
+  double curvature = std::clamp(icurvature, -1.0, 1.0);
+  if (std::abs(curvature) < ithreshold) {
+    curvature = 0;
+  }
+
+  // the algorithm switches to arcade when forward speed is 0 to allow point turns.
+  if(forwardSpeed == 0){
```

Comment:
Looks like there are a few code style issues that would be resolved by running `clang-format` per https://github.com/OkapiLib/OkapiLib/blob/master/CONTRIBUTING.md#clang-format

Reply:
I have a quick question on how to approach this. I have ran clang-format, but it seems like there were other files (such as asyncMotionProfileController) that were formatted as well.
![image](https://user-images.githubusercontent.com/71594512/172015356-b85a5f60-1d9b-4ca4-a0ba-73709d0d7e97.png)

Should I only commit the files that I've edited, or should I go ahead and commit all the files that were formatted.

Reply:
Well, I have gone ahead and modified all of them in the latest commit. The formatting issues should be fixed now.